src/main/java/net/snowflake/client/core/SessionUtil.java

```diff
@@ -1367,7 +1367,7 @@ static void updateSfDriverParamValues(Map<String, Object> parameters, SFSession
           session.setClientPrefetchThreads((int) entry.getValue());
         }
       } else if (CLIENT_OUT_OF_BAND_TELEMETRY_ENABLED.equalsIgnoreCase(entry.getKey())) {
-        if ((boolean) entry.getValue()) {
+        if ((boolean) entry.getValue() && session != null && session.isTelemetryEnabled()) {
```

Comment:
@mknister what do you think?

Reply:
Ah, this would prevent OOB telemetry from being sent whenever the session is null. This is used by the default connection handler, so I think this would stop some OOB telemetry. @sfc-gh-ema , what's the reason for this added logic? I think stopping OOB telemetry when the session is null should only apply to the non-default connection handler but this change will affect telemetry everywhere.

Reply:
Yeah, this is because we don't want to allow any telemetry being sent in our "other" JDBC implementation. I have the null check to avoid an NPE, and the current manner of checking whether we should support telemetry is via `isTelemetryEnabled` in `SFBaseSession` (true in `DefaultSFConnectionHandler`, but can be false). 

I realize that this may not be the best way to do it. Do you have any other suggestion for turning this off? Maybe via some sort of static variable?

Reply:
So the new plan, after discussing with @sfc-gh-mknister is to return a `NoOpTelemetryClient` in `getTelemetryClient`. To disable OOB telemetry we will issue an `alter session set CLIENT_OUT_OF_BAND_TELEMETRY_ENABLED = false` when we open the `StoredProcSFSession`.